pkg/cmd/server/origin/controller/autoscaling.go

```
@@ -18,9 +18,7 @@ type HorizontalPodAutoscalerControllerConfig struct {
 }
 
 func (c *HorizontalPodAutoscalerControllerConfig) RunController(originCtx ControllerContext) (bool, error) {
-	ctx := originCtx.KubeControllerContext
-
-	hpaClientConfig := ctx.ClientBuilder.ConfigOrDie(bootstrappolicy.InfraHorizontalPodAutoscalerControllerServiceAccountName)
+	hpaClientConfig := originCtx.ClientBuilder.ConfigOrDie(bootstrappolicy.InfraHorizontalPodAutoscalerControllerServiceAccountName)
```

we could return the error here, no need to die.